File: baekjoon/bronze/level2/test_2231/java/Main.java
package bronze.level2.test_2231.java;

import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());
        int result = 0;

        for(int i = 1; i <= n; i++){
            char[] c = String.valueOf(i).toCharArray();
            int sum = 0;
            for (char value : c) {
                sum += Integer.parseInt(String.valueOf(value));
            }

            if (n == sum + i) {
                result = n - sum;
                break;
            }
        }

        bw.write(String.valueOf(result));
        bw.close();
    }
}

File: baekjoon/bronze/level2/test_2231/kotlin/Main.kt
package bronze.level2.test_2231.kotlin

fun main() {
    val n = readln().toInt()
    var value = 1

    while (value < n) {
        val sumOfDigits = value.toString().sumOf { it.digitToInt() }
        if (sumOfDigits + value == n) {
            print(value)
            return
        }
        value++
    }

    print(0)
}
